File: Stacks/Python/Stack.py
import os
import sys
from Node import Node

class Stack:

    def __init__(self):
        self.Top = None

    def printStack(self):
        if self.Top == None:
            print("The stack is empty \n")
        else:
            aux = self.Top
            while aux != None:
                print(aux.data,end=" ")
                aux = aux.next
            
            print()
    
    def pushS(self,d):
        e = Node(d)
        
        if  self.Top == None:
            self.Top = e
        else:
            e.next = self.Top
            self.Top = e
    
    def popS(self):
        if(self.Top != None):
            self.Top = self.Top.next

    def isEmpty(self):
        if(self.Top == None):
            return True
        else:
            return False
    
    def peekS(self):
        if self.Top !=None:
            print("The top element in the stack is \n", self.Top.data)



"""Testing"""
S = Stack()

print("Give me the number of elements you want to push in the stack: \n")
N = int(input())

print("Give me your elements \n")
for i in range(N):
    e = int(input())
    S.pushS(e)


print()
print("The elements in the stack are: \n")
S.printStack()

print()

S.popS()

if S.isEmpty():
    print("The stack is empty \n")
else:
    print("the stack is not empty \n")

print("Your stack after popping \n")
S.printStack()

S.peekS()

S.popS()
S.popS()


if S.isEmpty():
    print("The stack is empty \n")
else:
    print("the stack is not empty \n")

print("Your stack after popping again\n")
S.printStack()

File: Stacks/Java/Stack.java
import java.io.*;
import java.util.Scanner;

class Stack<T>{
    
    Node<T> Top;

    public Stack(){
        this.Top = null;
    }

    public static <T>void printStack(Stack<T> s){
        if(s.Top == null)
        System.out.println("The stack is empty");
        else{
            Node<T> aux = s.Top;
            while(aux != null){
                System.out.print(aux.data+" ");
                aux = aux.next;
            }
        }
        System.out.println("\n");
    }

    public static <T>void pushS(Stack<T> s,T d){

            Node<T> n = new Node<>(d);
            if(s.Top == null)
            s.Top = n;
            else{
                n.next = s.Top;
                s.Top = n;
            }
            
    }

    public static <T>void popS(Stack<T> s){
        if(s.Top == null)
        System.out.println("The stack is empty");
        else s.Top = s.Top.next;
    }

    public static <T>Node<T> peekS(Stack<T> s){
        if(s.Top != null)
        return s.Top;
        else return null;
    }

    public static <T>boolean isEmpty(Stack<T> s){
        if(s.Top == null)
        return true;
        else return false;
    }

    public static void main(String args[]){

        Scanner sc = new Scanner(System.in);

        Stack<Integer> s = new Stack<>();
        int N;

        System.out.println("How many elemnet would you like to stack:");
        N = sc.nextInt();
        
        System.out.println("Give me your elements");
        int i = 0;
        while(i<N){
            int e = sc.nextInt();
            pushS(s,e);
            i++;
        }   

        System.out.println("");
        printStack(s);

        System.out.println("");
        
        Node<Integer> r = peekS(s);

        if(r != null)
        System.out.println("The top is: "+r.data);
        else System.out.println("There is no top");

        popS(s);
        popS(s);
        popS(s);

        if(isEmpty(s))
        System.out.println("It is empty \n");
        else    System.out.println("It is full \n");

        r = peekS(s);
        if(r != null)
        System.out.println("The top is: "+r.data);
        else System.out.println("There is no top");


        System.out.println("\n");
        printStack(s);
    }
}